### ISSUE TITLE
Problem reporting website: opts must include a valid :workspace.

### ISSUE THREAD

**Issue by oytunistrator** [rapid7/metasploit-framework#10999]
## Steps to reproduce
1. workspace -a lab1
2. load wmap
3. wmap_sites -a http://site.....

## Current behavior
```
[-] Error while running command wmap_sites: Problem reporting website: opts must include a valid :workspace. See log for more details.

Call stack:
/home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/core.rb:174:in `log_error'
/home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:8:in `rescue in report_web_site'
/home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:2:in `report_web_site'
/home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:1347:in `add_web_site'
/home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:132:in `cmd_wmap_sites'
/home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:502:in `run_command'
/home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:453:in `block in run_single'
/home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `each'
/home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `run_single'
/home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/shell.rb:151:in `run'
/home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
/home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
/home/oytun/pentest/exploits/metasploit-framework/msfconsole:49:in `<main>'
```

~/.msf4/logs$ cat framework.log;
```
[11/21/2018 11:49:09] [d(0)] core: Created user based module store
[11/21/2018 11:49:59] [e(0)] core: Problem reporting website: opts must include a valid :workspace
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/util/db_manager.rb:35:in `process_opts_workspace'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:191:in `block in report_web_site'
[11/21/2018 11:49:59] [e(0)] core: /var/lib/gems/2.5.0/gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:190:in `report_web_site'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:5:in `block in report_web_site'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/core.rb:166:in `data_service_operation'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:4:in `report_web_site'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:1347:in `add_web_site'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:132:in `cmd_wmap_sites'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:502:in `run_command'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:453:in `block in run_single'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `each'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `run_single'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/shell.rb:151:in `run'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
[11/21/2018 11:49:59] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/msfconsole:49:in `<main>'
[11/21/2018 11:50:22] [e(0)] core: Problem reporting website: opts must include a valid :workspace
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/util/db_manager.rb:35:in `process_opts_workspace'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:191:in `block in report_web_site'
[11/21/2018 11:50:22] [e(0)] core: /var/lib/gems/2.5.0/gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:190:in `report_web_site'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:5:in `block in report_web_site'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/core.rb:166:in `data_service_operation'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:4:in `report_web_site'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:1347:in `add_web_site'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:132:in `cmd_wmap_sites'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:502:in `run_command'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:453:in `block in run_single'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `each'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `run_single'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/shell.rb:151:in `run'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
[11/21/2018 11:50:22] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/msfconsole:49:in `<main>'
[11/21/2018 11:52:09] [e(0)] core: Problem reporting website: opts must include a valid :workspace
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/util/db_manager.rb:35:in `process_opts_workspace'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:191:in `block in report_web_site'
[11/21/2018 11:52:09] [e(0)] core: /var/lib/gems/2.5.0/gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:190:in `report_web_site'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:5:in `block in report_web_site'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/core.rb:166:in `data_service_operation'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:4:in `report_web_site'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:1347:in `add_web_site'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:132:in `cmd_wmap_sites'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:502:in `run_command'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:453:in `block in run_single'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `each'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `run_single'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/shell.rb:151:in `run'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
[11/21/2018 11:52:09] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/msfconsole:49:in `<main>'
[11/21/2018 11:55:38] [e(0)] core: Problem reporting website: opts must include a valid :workspace
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/util/db_manager.rb:35:in `process_opts_workspace'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:191:in `block in report_web_site'
[11/21/2018 11:55:38] [e(0)] core: /var/lib/gems/2.5.0/gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:190:in `report_web_site'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:5:in `block in report_web_site'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/core.rb:166:in `data_service_operation'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:4:in `report_web_site'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:1347:in `add_web_site'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:132:in `cmd_wmap_sites'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:502:in `run_command'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:453:in `block in run_single'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `each'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `run_single'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/shell.rb:151:in `run'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
[11/21/2018 11:55:38] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/msfconsole:49:in `<main>'
[11/21/2018 11:55:42] [e(0)] core: Problem reporting website: opts must include a valid :workspace
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/util/db_manager.rb:35:in `process_opts_workspace'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:191:in `block in report_web_site'
[11/21/2018 11:55:42] [e(0)] core: /var/lib/gems/2.5.0/gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:190:in `report_web_site'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:5:in `block in report_web_site'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/core.rb:166:in `data_service_operation'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:4:in `report_web_site'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:1347:in `add_web_site'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:132:in `cmd_wmap_sites'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:502:in `run_command'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:453:in `block in run_single'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `each'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `run_single'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/shell.rb:151:in `run'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
[11/21/2018 11:55:42] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/msfconsole:49:in `<main>'
[11/21/2018 11:55:44] [e(0)] core: Problem reporting website: opts must include a valid :workspace
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/util/db_manager.rb:35:in `process_opts_workspace'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:191:in `block in report_web_site'
[11/21/2018 11:55:44] [e(0)] core: /var/lib/gems/2.5.0/gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/msf/core/db_manager/web.rb:190:in `report_web_site'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:5:in `block in report_web_site'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/core.rb:166:in `data_service_operation'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/data_service/proxy/web_data_proxy.rb:4:in `report_web_site'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:1347:in `add_web_site'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/plugins/wmap.rb:132:in `cmd_wmap_sites'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:502:in `run_command'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:453:in `block in run_single'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `each'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:447:in `run_single'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/rex/ui/text/shell.rb:151:in `run'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
[11/21/2018 11:55:44] [e(0)] core: /home/oytun/pentest/exploits/metasploit-framework/msfconsole:49:in `<main>'
```

### Metasploit version
Framework Version: 5.0.0-dev-7084538b8c

### I installed Metasploit with:
- [ ] Kali package via apt
- [ ] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [x] Source install (please specify ruby version)

### OS
Ubuntu 18.10 Server Edition
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]

**Comment by threedr3am** [comment]
My mac os 10.13.6 have this problem too.

**Comment by exenin** [comment]
had the same issue - the fix is to pass the workspace to the report_website method
https://github.com/exenin/metasploit-framework/commit/863ffb338679a35d7a615276fa1408b02102c7d2

**Comment by bcoles** [comment]
Closed #10999 via #11284
